wagtail_newsletter/rich_text.py

```diff
@@ -0,0 +1,46 @@
+from copy import copy
+from functools import lru_cache
+
+from django.utils.html import escape
+from wagtail.models import Page
+from wagtail.rich_text import EmbedRewriter, LinkRewriter, MultiRuleRewriter, features
+from wagtail.rich_text.pages import PageLinkHandler
+
+
+class LinkHandlerForEmail(PageLinkHandler):
+    @classmethod
+    def expand_db_attributes(cls, attrs):
+        try:
+            page = cls.get_instance(attrs)
+            full_url = page.localized.specific.full_url  # type: ignore
+            return '<a href="%s">' % escape(full_url)
```

Comment:
This syntax being constantly referred to as `the old %-formatting` in the Python doc, maybe we'd rather use its more modern `.format()` form, or an f-string? :slightly_smiling_face:

Reply:
Weeelll, I just [stole that code from Wagtail](https://github.com/wagtail/wagtail/blob/4f5ffa85b660bb76c5db43e72d5b6f37b884fb06/wagtail/rich_text/pages.py#L33), but you're right, might as well fix it.

Interestingly, my editor's ruff complained about this, but not the pre-commit hook; turns out the hook is running a slightly outdated version ([v0.3.0](https://github.com/astral-sh/ruff-pre-commit/releases/tag/v0.3.0)), so I bumped it to latest ([v0.4.4](https://github.com/astral-sh/ruff-pre-commit/releases/tag/v0.4.4)).